src/amo/components/SearchForm/index.js

```diff
@@ -0,0 +1,215 @@
+
+    const { addonType, dispatch, errorHandler } = this.props;
+    const filters = { query: value };
+
+    if (addonType) {
+      filters.addonType = addonType;
+    }
+
+    dispatch(autocompleteStart({
+      errorHandlerId: errorHandler.id,
+      filters,
+    }));
+  }
+
+  handleSuggestionsClearRequested = () => {
+    this.props.dispatch(autocompleteCancel());
+  }
+
+  handleSuggestionSelected = (e, { suggestion }) => {
+    e.preventDefault();
+    this.setState({ searchValue: '' }, () => {
+      this.context.router.push(suggestion.url);
+    });
+  }
+
+  renderSuggestion = (suggestion) => {
+    if (this.props.loadingSuggestions) {
+      return <LoadingText width={60} />;
+    }
+
+    const { name, iconUrl } = suggestion;
+
+    return <Suggestion name={name} iconUrl={iconUrl} />;
+  }
+
+  render() {
+    const { addonType, api, i18n, pathname } = this.props;
+
+    let placeholderText;
+    if (addonType === ADDON_TYPE_EXTENSION) {
+      placeholderText = i18n.gettext('Search extensions');
+    } else if (addonType === ADDON_TYPE_THEME) {
+      placeholderText = i18n.gettext('Search themes');
+    } else {
+      placeholderText = i18n.gettext('Search extensions and themes');
+    }
+
+    return (
+      <form
+        method="GET"
+        action={`/${api.lang}/${api.clientApp}${pathname}`}
+        onSubmit={this.handleSearch}
+        className="SearchForm-form"
+        ref={(ref) => { this.form = ref; }}
+      >
+        <label
+          className="visually-hidden"
+          htmlFor="q"
+        >
+          {i18n.gettext('Search')}
+        </label>
+        <Autosuggest
+          className="SearchForm-suggestions"
+          focusInputOnSuggestionClick={false}
+          getSuggestionValue={(suggestion) => suggestion.name}
+          inputProps={{
+            value: this.state.searchValue,
+            onChange: this.handleSearchChange,
+            placeholder: placeholderText,
+            className: 'SearchForm-query',
+            name: 'q',
+            type: 'search',
+            inputRef: (ref) => { this.searchInput = ref; },
+          }}
+          onSuggestionsClearRequested={this.handleSuggestionsClearRequested}
+          onSuggestionsFetchRequested={this.handleSuggestionsFetchRequested}
+          onSuggestionSelected={this.handleSuggestionSelected}
+          renderSuggestion={this.renderSuggestion}
+          renderInputComponent={(inputProps) => <SearchInput {...inputProps} />}
+          suggestions={this.getSuggestions()}
+          theme={{
+            suggestionContainer: 'SearchForm-suggestions',
+            suggestionsList: 'SearchForm-suggestions-list',
+            suggestion: 'SearchForm-suggestions-item',
+            suggestionHighlighted: 'SearchForm-suggestions-item--highlighted',
+          }}
+        />
+        <button
+          className="visually-hidden"
+          onClick={this.handleSearch}
+          ref={(ref) => { this.submitButton = ref; }}
+          title="Enter"
```

Comment:
Even though it's hidden, this should be localized like `title={this.props.i18n.gettext('Enter')}` for accessibility. When you move old code around you get to own it 🌵